src/commands/selection.rs

```diff
@@ -5,33 +5,56 @@ use crate::errors::*;
 use crate::commands::{self, Result};
 use crate::util;
 
+use regex::Regex;
+
 pub fn delete(app: &mut Application) -> Result {
-    if let Some(buffer) = app.workspace.current_buffer() {
-        match app.mode {
-            Mode::Select(ref select_mode) => {
-                let cursor_position = *buffer.cursor.clone();
-                let delete_range = Range::new(cursor_position, select_mode.anchor);
-                buffer.delete_range(delete_range.clone());
-                buffer.cursor.move_to(delete_range.start());
-            }
-            Mode::SelectLine(ref mode) => {
-                let delete_range = mode.to_range(&*buffer.cursor);
-                buffer.delete_range(delete_range.clone());
-                buffer.cursor.move_to(delete_range.start());
-            }
-            Mode::Search(ref mode) => {
-                let selection = mode.results
-                    .as_ref()
-                    .and_then(|r| r.selection())
-                    .ok_or("Can't delete in search mode without a selected result")?;
-                buffer.delete_range(selection.clone());
-            }
-            _ => bail!("Can't delete selections outside of select mode"),
-        };
-    } else {
+    if app.workspace.current_buffer().is_none() {
+        bail!(BUFFER_MISSING);
+    }
+    match app.mode {
+        Mode::Select(_) | Mode::SelectLine(_) | Mode::Search(_) => {
+            let delete_range = range_from(app);
+            let buffer = app.workspace.current_buffer().unwrap();
+
+            buffer.delete_range(delete_range.clone());
+            buffer.cursor.move_to(delete_range.start());
+        }
+        _ => bail!("Can't delete selections outside of select mode"),
+    };
+
+    Ok(())
+}
+
+pub fn justify(app: &mut Application) -> Result {
+    if app.workspace.current_buffer().is_none() {
         bail!(BUFFER_MISSING);
     }
 
+    let range = match app.mode {
+        Mode::Select(_) | Mode::SelectLine(_) | Mode::Search(_) => {
+            range_from(app)
+        }
+        _ => bail!("Can't justify without selection"),
+    };
+
+    // delete and save the range, then justify that range
+    let buffer = app.workspace.current_buffer().unwrap();
+    if let Some(text) = buffer.read(&range.clone()) {
+        buffer.delete_range(range.clone());
+        buffer.cursor.move_to(range.start());
+
+        buffer.insert(
+            &justify_string(
+                &text,
+                match app.preferences.borrow().line_length_guide() {
+                    Some(l) => l,
+                    None => 80,
+                },
+                Regex::new(r"#|//|/\*|///").unwrap(),
+            )
+        );
+    }
+
     Ok(())
```

Comment:
It should switch back to `normal` mode after reflowing, at least that's what I expect when using this command.
```suggestion
    application::switch_to_normal_mode(app)
```

Reply:
I'd agree with this suggestion. 🙂